x-pack/plugins/transform/public/app/hooks/use_delete_transform.tsx

```diff
@@ -4,52 +4,273 @@
-
-import { getErrorMessage } from '../../shared_imports';
-
+import { TransformEndpointRequest, DeleteTransformEndpointResult } from '../../../common';
+import { getErrorMessage, extractErrorMessage } from '../../shared_imports';
 import { useAppDependencies, useToastNotifications } from '../app_dependencies';
 import { TransformListRow, refreshTransformList$, REFRESH_TRANSFORM_LIST_STATE } from '../common';
 import { ToastNotificationText } from '../components';
-
 import { useApi } from './use_api';
+import { API_BASE_PATH } from '../../../common/constants';
+import { IIndexPattern } from '../../../../../../src/plugins/data/common/index_patterns';
+
+export const canDeleteIndex = async (http: CoreSetup['http']) => {
+  const privilege = await http.get(`${API_BASE_PATH}privileges`);
+  if (!privilege) {
+    return false;
+  }
+  return privilege.hasAllPrivileges;
+};
+
+export const useDeleteIndexAndTargetIndex = (items: TransformListRow[]) => {
+  const { http, savedObjects } = useAppDependencies();
+  const toastNotifications = useToastNotifications();
+
+  const [deleteDestIndex, setDeleteDestIndex] = useState<boolean>(true);
+  const [deleteIndexPattern, setDeleteIndexPattern] = useState<boolean>(true);
+  const [userCanDeleteIndex, setUserCanDeleteIndex] = useState<boolean>(false);
+  const [indexPatternExists, setIndexPatternExists] = useState<boolean>(false);
+  const toggleDeleteIndex = useCallback(() => setDeleteDestIndex(!deleteDestIndex), [
+    deleteDestIndex,
+  ]);
+  const toggleDeleteIndexPattern = useCallback(() => setDeleteIndexPattern(!deleteIndexPattern), [
+    deleteIndexPattern,
+  ]);
+  const savedObjectsClient = savedObjects.client;
+
+  const checkIndexPatternExists = useCallback(
+    async (indexName: string) => {
+      try {
+        const response = await savedObjectsClient.find<IIndexPattern>({
+          type: 'index-pattern',
+          perPage: 10,
+          search: `"${indexName}"`,
+          searchFields: ['title'],
+          fields: ['title'],
+        });
+        const ip = response.savedObjects.find(
+          (obj) => obj.attributes.title.toLowerCase() === indexName.toLowerCase()
+        );
+        if (ip !== undefined) {
+          setIndexPatternExists(true);
+        }
+      } catch (e) {
+        const error = extractErrorMessage(e);
+
+        toastNotifications.addDanger(
+          i18n.translate(
+            'xpack.transform.deleteTransfrom.errorWithCheckingIfIndexPatternExistsNotificationErrorMessage',
+            {
+              defaultMessage:
+                'An error occurred checking if index pattern {indexPattern} exists: {error}',
+              values: { indexPattern: indexName, error },
+            }
+          )
+        );
+      }
+    },
+    [savedObjectsClient, toastNotifications]
+  );
+
+  const checkUserIndexPermission = useCallback(() => {
+    try {
+      const userCanDelete = canDeleteIndex(http);
```

Comment:
I this needs an `await`.

Reply:
Fixed https://github.com/elastic/kibana/pull/67922/commits/ec3acd3ffcec0cdf17fa414a355208ca0e5ce9f8